syntax = "proto3";

package fetch.oef.pb;

message CoreConfig {
    string core_key = 1;
    string core_uri = 2;
    string search_uri = 3;
    string ws_uri = 4;

    uint32 comms_thread_count = 10;
    uint32 tasks_thread_count = 11;

    map<string, string> karma_policy = 20;
    uint32 karma_refresh_interval_ms = 21;

    string prometheus_log_file = 30;
    uint32 prometheus_log_interval = 31;
    
    string ssl_uri = 40;
    string secure_uri = 41;
    string white_list_file = 42;
    string core_cert_pk_file = 43;
    string core_pubkey_file = 44;
    string tmp_dh_file = 45;
}
